src/components/devops-problem.astro

```diff
@@ -0,0 +1,52 @@
+---
+import { z } from 'zod';
+import { client } from '../sanity-client.ts';
+
+const problems = z
+  .array(z.string())
+  .parse(await client.fetch(`*[_type == 'problem'].problem`));
+---
+
+<devops-problem data-problems={JSON.stringify(problems)}></devops-problem>
+
+<script>
+  import { z } from 'zod';
+
+  class DevopsProblem extends HTMLElement {
```

Comment:
I _think_ a web component is the cleanest way to pass state from the frontmatter to a client-side script. I don't love it, but it works.